### PR TITLE
feat: updated desktop modeler plugin documentation

### PR DESCRIPTION
**Pull request by Skaiir**
Closes https://github.com/camunda/camunda-modeler/issues/2782

### CODE REVIEW

File: docs/components/modeler/desktop-modeler/plugins/plugins.md [hunk 1]
```diff
@@ -4,15 +4,39 @@ title: Plugins
...
-So let's dive into how to add your own plugins.
+<br>
+
+## Overview of your possibilities as a plugin developer
+
+There are many way you as a developer are able to extend our modeling tools, and since this ecosystem is a little large here's an overview of the ways you can customize our software:
+
+| Plugin type               | Functionality                                                                                                        | Example                                                                                                                         |
+| ------------------------- | -------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| Menu Entries              | Add new entries to the menu bar - useful to interact with your plugins, link to external pages, add settings, ect... | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/camunda/camunda-modeler-plugins/tree/master/menu-plugin-example)                     |
```
[the remark this answers is not in the document]
Comment (MaxTru) [reply docs/components/modeler/desktop-modeler/plugins/plugins.md:29]: This applies to all of the following lines

File: docs/components/modeler/desktop-modeler/plugins/plugins.md [hunk 2]
```diff
@@ -4,15 +4,39 @@ title: Plugins
...
+| bpmn-js Modules           | Extend our BPMN editor by creating your own custom modules                                                           | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/camunda/camunda-modeler-plugins/tree/master/bpmn-js-plugin-example)                  |
+| bpmn-js Moddle Extensions | Enable the use of custom BPMN semantics                                                                              | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/camunda/camunda-modeler-plugins/tree/master/bpmn-js-plugin-moddle-extension-example) |
+| dmn-js Modules            | Extend our DMN editor by creating your own custom modules                                                            | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/camunda/camunda-modeler-plugins/tree/master/dmn-js-plugin-example)                   |
+| dmn-js Moddle Extensions  | Enable the use of custom DMN semantics                                                                               | &nbsp;&nbsp;&nbsp;n/a |
```
Comment (MaxTru) [inline docs/components/modeler/desktop-modeler/plugins/plugins.md:35]:
```suggestion
| dmn-js Moddle Extensions  | Extend the DMN language model by injecting your own custom [dmn-moddle](https://github.com/bpmn-io/dmn-moddle) modules                                                                             | &nbsp;&nbsp;&nbsp;n/a |
```

File: docs/components/modeler/desktop-modeler/plugins/plugins.md [hunk 3]
```diff
@@ -4,15 +4,39 @@ title: Plugins
...
+| ------------------------- | -------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| Menu Entries              | Add new entries to the menu bar - useful to interact with your plugins, link to external pages, add settings, ect... | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/camunda/camunda-modeler-plugins/tree/master/menu-plugin-example)                     |
+| Custom Styles             | Change the look and feel of Camunda Modeler and your extensions                                                      | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/camunda/camunda-modeler-plugins/tree/master/style-plugin-example)                    |
+| React Components          | Embed custom React components into specific anchor points in our application                                         | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/pinussilvestrus/camunda-modeler-autosave-plugin)                                     |
```
Comment (MaxTru) [inline docs/components/modeler/desktop-modeler/plugins/plugins.md:31]: Sometimes you talk about `our application` sometimes about `Camunda Modeler`. Please make it consistent. I recommend:
```suggestion
| React Components          | Embed custom React components into specific anchor points of the Camunda Modeler                                       | &nbsp;&nbsp;&nbsp;[**↗**](https://github.com/pinussilvestrus/camunda-modeler-autosave-plugin)                                     |
```

This also applies to the other table entries

File: docs/components/modeler/desktop-modeler/plugins/plugins.md [hunk 4]
```diff
@@ -23,9 +47,10 @@ module.exports = {
   name: 'My Awesome Plugin', // the name of your plugin
   style: './style.css', // changing the appearance of the modeler
   menu: './menu.js', // adding menu entries to the modeler
-  script: './script.js' // extend the modeling tools for BPMN, or DMN
+  script: './script.js' // extend the modeler, and its BPMN and DMN components
```
Comment (MaxTru) [inline docs/components/modeler/desktop-modeler/plugins/plugins.md:50]:
```suggestion
  script: './script.js' // extending the modeler, and its BPMN and DMN components
```

File: docs/components/modeler/desktop-modeler/plugins/plugins.md [hunk 5]
```diff
@@ -128,6 +155,13 @@ var plugin = require('./LoggingPlugin');
 registerBpmnJSPlugin(plugin);
 ```
 
+You may want to create a plugin which specifically targets Camunda 7 or Camunda 8. To do this use the appropriate variations of the registration helper function for your plugin type.
+
+```javascript
+// Such variations are currently only available for BPMN but will soon become available for DMN as well
+registerPlatformBpmnJSPlugin(plugin)
```
Comment (MaxTru) [inline docs/components/modeler/desktop-modeler/plugins/plugins.md:162]:
```suggestion
registerPlatformBpmnJSPlugin(plugin); // Register plugin for Camunda 7 BPMN diagrams only
registerCloudBpmnJSPlugin(plugin); // Register plugin for Camunda 8 BPMN diagrams only
registerBpmnJSPlugin(plugin); // Register plugin for Camunda 7 and 8 BPMN diagrams
```

**Review by MaxTru — CHANGES_REQUESTED** [review]

File: docs/components/modeler/desktop-modeler/plugins/plugins.md [hunk 6]
```diff
@@ -76,28 +93,30 @@ Plug them into the modeler like this:
 
 ```javascript
 module.exports = {
-  menu: './menu-entries'
+  menu: './menu-entries'
 };
 ```
 
-You can use a Node.js module here since the modeler is built with [Electron](https://electron.atom.io/) which uses [Node.js](https://nodejs.org/en/).
+> The code within the menu entries is special because it executes on [the main process](https://www.electronjs.org/docs/latest/tutorial/process-model) of Electron. This comes with the advantage of allowing you to use [Node.js](https://nodejs.org/en/) modules, and the disadvantage of not being able to debug your code in Chromium. For more information regarding main process debugging, please refer to the [official Electron documentation](https://www.electronjs.org/docs/latest/tutorial/debugging-main-process).
```
Comment (Skaiir) [inline docs/components/modeler/desktop-modeler/plugins/plugins.md:100]: @MaxTru Added this entry, thoughts?

Comment (MaxTru) [reply docs/components/modeler/desktop-modeler/plugins/plugins.md:100]: I like. 

I would propose a slight rephrasing (not a disadvantage, but a challenge to consider ;-) ). Something like:

```suggestion
> The code within the menu entries executes on [the main process](https://www.electronjs.org/docs/latest/tutorial/process-model) of Electron. This comes with the advantage of allowing you to use [Node.js](https://nodejs.org/en/) modules, but you need to consider, that you can not debug the respective code in Chromium. For more information regarding main process debugging, please refer to the [official Electron documentation](https://www.electronjs.org/docs/latest/tutorial/debugging-main-process).
```

**Review by MaxTru — APPROVED** [review]
Good job! :wave:

**Comment by akeller** [comment]
@christinaausley can you review?

File: docs/components/modeler/desktop-modeler/plugins/plugins.md [hunk 7]
```diff
@@ -76,35 +97,41 @@ Plug them into the modeler like this:
...
+### Extend the modeler and its BPMN and DMN components
 
-### Extend the Modeling Tools for BPMN, and DMN
+You can extend the modeling tools for [BPMN](https://github.com/bpmn-io/bpmn-js) and [DMN](https://github.com/bpmn-io/dmn-js) with your own modules, as well as embedding React.js components into certain sections of Camunda Modeler.
 
-> Currently you can only extend bpmn-js and dmn-js
+Since the client of the modeler uses [Chromium](https://www.chromium.org/Home), you can't use Node.js modules to extend the modeling tools. You need to bundle your plugin first. The easiest way to get started with client-side plugins is through [this example project](https://github.com/camunda/camunda-modeler-plugin-example).
 
-You can extend the modeling tools for [BPMN](https://github.com/bpmn-io/bpmn-js), and [DMN](https://github.com/bpmn-io/dmn-js) with your own modules.
+> In this example, we are building a bpmn-js plugin, but this basic structure applies to all extensions besides menu entries and style. The modules themselves will be different however, so refer to our [examples](https://github.com/camunda/camunda-modeler-plugins) for more information on how to build different kinds.
```
Comment (christinaausley) [inline docs/components/modeler/desktop-modeler/plugins/plugins.md:116]: "...how to build different kinds of modules?"

**Review by christinaausley — APPROVED** [review]
Some tweaks made in my latest commit but this looks great -- very clean. I made one comment for clarity purposes. A few adjustments I made in the commit include:

1. Using plugin consistently as opposed to plug-in.
2. Using "Camunda Modeler" as opposed to "The Camunda Modeler."
3. Sentence-case headers throughout the document.
4. Applying steps where applicable to improve the user experience.
5. Applying notes on important concepts the user should keep in mind.

**Review by MaxTru — APPROVED** [review]

**Comment by MaxTru** [comment]
> Some tweaks made in my latest commit but this looks great -- very clean. I made one comment for clarity purposes. A few adjustments I made in the commit include:
> 
> 1. Using plugin consistently as opposed to plug-in.
> 2. Using "Camunda Modeler" as opposed to "The Camunda Modeler."
> 3. Sentence-case headers throughout the document.
> 4. Applying steps where applicable to improve the user experience.
> 5. Applying notes on important concepts the user should keep in mind.

Thanks!

**Pull request merged**